### PR TITLE
docs: Add Architecture documentation

### PR DESCRIPTION
**Pull request by titanous**

### CODE REVIEW

File: docs/content/architecture.html.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+where to send client traffic.
+
+## Buildpacks
+
+Flynn uses [Heroku Buildpacks](https://devcenter.heroku.com/articles/buildpacks)
+to convert application source code into a runnable artifact called a *slug*.
+This slug is a tarball of the app and all of it's dependencies, which can be run
+by the *slugrunner* component.
+
+Flynn is designed to be support a variety of deployment pipelines, so the
```
Comment (jvatic) [inline docs/content/architecture.html.md]: `s/to be support/to support/`

File: docs/content/architecture.html.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+buildpack support is not special or hard-coded, it just uses controller APIs to
+do the deploy.
+
+### gitreceive
+
+Git pushes are received over HTTPS by the gitreceive component which spawns
+a receiver that starts a new *slugbuilder* job to process the build. If the repo
+has been pushed before, a cached archive of the repo downloaded from the
```
Comment (jvatic) [inline docs/content/architecture.html.md]: `s/repo downloaded/repo will be downloaded/`

File: docs/content/architecture.html.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+Slugrunner is responsible for running a specific process type from an app slug
+created by slugbuilder. When an application deployed with the buildpack flow is
+scaled up, a slugrunner job is started for each process and then executes the
+application from the slug.
+
+## Log Aggregator
+
+The logaggregator receives syslog-formatted streams with log lines from jobs
+running on each flynn-host instance containing and buffers recent log lines for
```
Comment (jvatic) [inline docs/content/architecture.html.md]: This part of the sentence is hard to read

File: docs/content/architecture.html.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+used.
+
+JSON over HTTP is used for communication, both internal and external. Our
+experience is that it is easy to debug and understand HTTP-based services. We
+use [Server-Sent Events](https://www.w3.org/TR/eventsource/) to provide streams
+over HTTP. Since most developers are already familiar with HTTP and almost every
+language already has great HTTP support, taking advantage of Flynn's APIs is
+easier than yet another RPC protocol.
+
+We target the latest LTS release of Ubuntu amd64 as our base operating system.
```
Comment (lmars) [inline docs/content/architecture.html.md]: The latest LTS version is changing in the next few days, should we be more specific about which release we are using, or just switch to the latest release soon after it becomes available?

File: docs/content/architecture.html.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+## flannel
+
+An overlay network is automatically configured using the Linux kernel's native
+support for VXLAN. Each host in the cluster is assigned an /24 block of IPv4
+addresses and registered in discoverd.
+
+IPs from the block are allocated to each job that is started by flynn-host.
+Routes are configured by flannel that route all inter-container communication to
+the correct host.
```
Comment (lmars) [inline docs/content/architecture.html.md]: Is it worth mentioning that the packets travel via UDP?

**Comment by titanous** [comment]
Comments addressed.

**Comment by lmars** [comment]
LGTM

**Pull request merged**